### PR TITLE
8810 dont update solr names DB for SP and GP

### PR DESCRIPTION
**Pull request by ozamani9**
*Issue # https://github.com/bcgov/entity/issues/8810

*Description of changes:*

When nr is processed through the solr names updater bypass the posts to SOLR database functions if the company entity_type_cd is FR (Sole prop) or GP (General Partnership)

### CODE REVIEW

File: services/solr-names-updater/src/solr_names_updater/worker.py [hunk 1]
```diff
@@ -66,7 +67,9 @@ def is_names_event_msg_type(msg: dict):
 
 def is_processable(msg: dict):
     """Determine if message is processable using message type of msg."""
-    if msg and is_names_event_msg_type(msg):
+    nr_num = msg.get('nrNum', None)
+    nr = RequestDAO.find_by_nr(nr_num)
+    if msg and is_names_event_msg_type(msg) and nr.entity_type_cd not in ('FR', 'GP'):
```
Comment (thorwolpert) [inline services/solr-names-updater/src/solr_names_updater/worker.py:72]: Not saying you need to make this change, but there is a small chance (super small) the NR is gone, but it gives me a chance to show you the walrus (goo goo ga joob) ```:=```
```suggestion
    if msg and is_names_event_msg_type(msg) \
       and (nr := RequestDAO.find_by_nr(nr_num)) \
       and nr.entity_type_cd not in ('FR', 'GP'):
```

File: services/solr-names-updater/tests/unit/test_worker.py [hunk 2]
```diff
@@ -12,10 +12,52 @@
...
+     ('Gen Part Cancelled', create_request_state_change_message('CANCELLED', 'DRAFT'), 'GP', False),
+     ('Gen Part Reset', create_request_state_change_message('RESET', 'DRAFT'), 'GP', False),
+     ('Gen Part Consumed', create_request_state_change_message('CONSUMED', 'DRAFT'), 'GP', False),
+     ('CORPORATION Approved', create_request_state_change_message('APPROVED', 'DRAFT'), 'CR', True),
+     ('CORPORATION Conditional', create_request_state_change_message('CONDITIONAL', 'DRAFT'), 'CR', True),
+     ('CORPORATION Cancelled', create_request_state_change_message('CANCELLED', 'DRAFT'), 'CR', True),
+     ('CORPORATION Reset', create_request_state_change_message('RESET', 'DRAFT'), 'CR', True),
+     ('CORPORATION Consumed', create_request_state_change_message('CONSUMED', 'DRAFT'), 'CR', True)
+    ])
+async def test_sp_gp_names_not_processed_to_solr(
```
Comment (thorwolpert) [inline services/solr-names-updater/tests/unit/test_worker.py:47]: yay, awesome to add to the test suite!

**Review by thorwolpert — APPROVED** [review]
LGTM - A suggested change to add to your repertoire

File: services/solr-names-updater/src/solr_names_updater/worker.py (hunk 1 above)
Comment (ozamani9) [reply services/solr-names-updater/src/solr_names_updater/worker.py]: done.

File: services/solr-names-updater/src/solr_names_updater/worker.py [hunk 3]
```diff
@@ -66,7 +67,11 @@ def is_names_event_msg_type(msg: dict):
 
 def is_processable(msg: dict):
     """Determine if message is processable using message type of msg."""
-    if msg and is_names_event_msg_type(msg):
+    nr_num = msg.get('nrNum', None)
+    nr = RequestDAO.find_by_nr(nr_num)
```
Comment (argush3) [inline services/solr-names-updater/src/solr_names_updater/worker.py:71]: `nr = RequestDAO.find_by_nr(nr_num)` can be removed here as the `if` block will have it

Comment (ozamani9) [reply services/solr-names-updater/src/solr_names_updater/worker.py]: done

**Review by argush3 — APPROVED** [review]

**Review by argush3 — APPROVED** [review]
looks good!

**Pull request merged**